File: DataStructures/GeneralProblems/01_string_count.py
class StringCount(object):

    def __init__(self, string):
        self.string = string
        self.s = set()
        self.count = 0

    def getCount(self):
        for i in range(len(self.string)):
            subStr = self.string[i:]
            # print(f"subStr = {subStr}")
            for j in range(len(subStr)):
                sub = subStr[:len(subStr) - j]
                # print(f"sub = {sub}")
                if self.s.__contains__(sub):
                    continue
                self.count += 1
                self.s.add(sub)

        return self.count, self.s


strCount = StringCount("abcdelsadkjfowiqeuropiwqurzwuyeiuqweyoirurtpqcvbklasdjfhioweryoqiweuryrnbmvcnbzxmcnvbajkhgeriytqowieuryoiquwyralskjdhfzmcnvbmz,nvbjhgasdfpqeurpqweroiuweryoiuytiudhfajshfmnbcvzmcxbzmcnbalkdjheoeyrtqutrjhgafeasdfasdfasddsfdfgsdgrteeytyerytrtyiuyityiubncvbncvbbmcmdfsgagagdfggkl")
count, s = strCount.getCount()
print(f"{count}")

File: DataStructures/GeneralProblems/01_string_count.java
// Java Code

import java.util.*;

public class Main{

     public static void main(String []args) {

        Set<String> s = new HashSet<String>();
        String str = "abcde";
        long count = 0;
        for (int i = 0; i < str.length(); i++) {
            String sub = str.substring(i);
            for (int j = 0; j < sub.length(); j++) {
                String subStr = sub.substring(0, sub.length() - j);
                if (s.contains(subStr)) {
                    continue;
                }
                count++;
                s.add(subStr);
            }
        }
        System.out.println(s);
        System.out.println(count);

     }
}